protocol/cosi.go

```diff
@@ -147,44 +149,38 @@ func (pc *ProtocolCosi) Dispatch() error {
 }
 
 // StartAnnouncement will start a new announcement.
-func (pc *ProtocolCosi) StartAnnouncement() error {
+func (pc *CoSi) StartAnnouncement() error {
```

Comment:
Can you merge this function with `handleAnnouncement`, so that if `handleAnnouncement` gets a `nil`, it will automatically start as a root-node?